src/plugins/Polls/UI/PollsDialog.tsx

```diff
@@ -0,0 +1,348 @@
+import React, { useState, useEffect } from 'react'
+import {
+    makeStyles,
+    createStyles,
+    DialogTitle,
+    DialogContent,
+    DialogProps,
+    Typography,
+    IconButton,
+    Button,
+    InputLabel,
+    TextField,
+    FormControl,
+    Select,
+    MenuItem,
+    Divider,
+    CircularProgress,
+} from '@material-ui/core'
+import AddIcon from '@material-ui/icons/Add'
+import ShadowRootDialog from '../../../utils/jss/ShadowRootDialog'
+import { PortalShadowRoot } from '../../../utils/jss/ShadowRootPortal'
+import { DialogDismissIconUI } from '../../../components/InjectedComponents/DialogDismissIcon'
+import { useStylesExtends, or } from '../../../components/custom-ui-helper'
+import AbstractTab, { AbstractTabProps } from '../../../extension/options-page/DashboardComponents/AbstractTab'
+import Services from '../../../extension/service'
+import { getActivatedUI } from '../../../social-network/ui'
+import { useCurrentIdentity } from '../../../components/DataSource/useActivatedUI'
+import type { PollGunDB } from '../Services'
+import { PollCardUI } from './Polls'
+import type { PollMetaData } from '../types'
+
+const useNewPollStyles = makeStyles((theme) =>
+    createStyles({
+        line: {
+            display: 'flex',
+            margin: theme.spacing(1),
+        },
+        item: {
+            flex: 1,
+            margin: theme.spacing(1),
+        },
+        pollWrap: {
+            border: '1px solid #ccd6dd',
+            borderRadius: '10px',
+            margin: theme.spacing(1),
+            padding: theme.spacing(1),
+        },
+        optionsWrap: {
+            position: 'relative',
+            '& >div': {
+                width: '80%',
+                margin: theme.spacing(2),
+            },
+        },
+        addButton: {
+            position: 'absolute',
+            bottom: '0',
+            right: '10px',
+        },
+        loading: {
+            color: '#fff',
+        },
+    }),
+)
+
+interface NewPollProps {
+    loading: [boolean, React.Dispatch<React.SetStateAction<boolean>>]
+    senderName: string | undefined
+    senderId: string | undefined
```

Comment:
What is `senderId`? Does it the serialized version of `ProfileIdentifier`?

Reply:
I see the code, it's the fingerprint of the persona. Please reflect it in the name.